app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/AacRecorder.kt

```diff
@@ -58,13 +62,23 @@ class AacRecorder {
         newData = newData.plus(bytesToShorts(array).toTypedArray())
         rawData = rawData.plus(array.toTypedArray())
 
+        if (newData.size >= NoiseDetector.DATA_SIZE &&
+            voiceAnalysisOption == VoiceAnalysisOption.NoiseDetection
+        ) {
+            soundDetectionData.onNext(newData.takeLast(NoiseDetector.DATA_SIZE).toShortArray())
+        }
+
         if (newData.size >= MachineLearning.DATA_SIZE) {
-            data.onNext(rawData.toByteArray() to newData.toShortArray())
-            newData = emptyArray()
-            rawData = emptyArray()
+            machineLearningData.onNext(rawData.toByteArray() to newData.toShortArray())
+            clearRecordingData()
         }
     }
 
+    private fun clearRecordingData() {
+        newData = emptyArray()
+        rawData = emptyArray()
+    }
+
     @RunsInBackground
     private fun bytesToShorts(bytes: ByteArray): ShortArray {
         val shorts = ShortArray(bytes.size / 2)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/8b49d288c8cda524f66973edfcfe165e7a30812e/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/AacRecorder.kt#L84"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>